### PR TITLE
Improve and unify testing facilities

### PR DESCRIPTION
**Pull request by bkchr**
This improves the testing facilities by making the test client easier to
use. It also removes code that is not required for the test client.
Besides that it also moves the test service and test client under
`node/test`.

### CODE REVIEW

File: node/test/client/src/block_builder.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	fn init_polkadot_block_builder<'a>(&'a self) -> sc_block_builder::BlockBuilder<'a, Block, Client, FullBackend>;
+
+	/// Init a Polkadot specific block builder at a specific block that works for the test runtime.
+	///
+	/// Same as [`InitPolkadotBlockBuilder::init_polkadot_block_builder`] besides that it takes a [`BlockId`] to say
+	/// which should be the parent block of the block that is being build.
+	fn init_polkadot_block_builder_at<'a>(
+		&'a self,
+		at: &BlockId<Block>,
+	) -> sc_block_builder::BlockBuilder<'a, Block, Client, FullBackend>;
```
Comment (ordian) [inline node/test/client/src/block_builder.rs:39]: are the lifetime annotations really required?

File: node/test/client/src/block_builder.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			.expect("Creates new block builder for test runtime");
+
+		let mut inherent_data = sp_inherents::InherentData::new();
+		let last_timestamp = self
+			.runtime_api()
+			.get_last_timestamp(&at)
+			.expect("Get last timestamp");
+
+		// `MinimumPeriod` is a storage parameter type that requires externalities to access the value.
+		let minimum_period= BasicExternalities::new_empty()
```
Comment (ordian) [inline node/test/client/src/block_builder.rs:64]:
```suggestion
		let minimum_period = BasicExternalities::new_empty()
```

File: node/test/client/src/block_builder.rs (hunk 1 above)
Comment (bkchr) [reply node/test/client/src/block_builder.rs]: Ty, it works without the explicit lifetimes. I always forget that this is now possible :)

**Review by montekki — APPROVED** [review]

**Comment by montekki** [comment]
bot merge

**Pull request merged**